File: easy/FindNUniqueInts.js
// Given an integer n, return any array containing n unique integers such that they add up to 0.

const sumZero = function (n) {
  const res = []
  let i = 0
  if (n % 2 !== 0) {
    res.push(0)
    i++
  }
  while (i !== n) {
    res.push(i + 1, -(i + 1))
    i += 2
  }
  return res
};

const ans = sumZero(25)
console.log(ans.reduce((partialSum, a) => partialSum + a, 0) === 0)

File: easy/FindNUniqueInts.py
# Given an integer n, return any array containing n unique integers such that they add up to 0.

class Solution:
    def sumZero(self, n: int):
        res = []
        if n % 2 == 1:
            res.append(0)
            n -= 1
        while n > 0:
            res.extend([n, -n])
            n -= 2
        return res


test = Solution()
ans = test.sumZero(6)
print(sum(ans) == 0 and len(ans) == 6)
